### ISSUE TITLE
SentinelOne Quarantines AltTab for persistence_deception

### ISSUE THREAD

**Issue by Davidian1024** [lwouis/alt-tab-macos#2725]
I recently switched to a MacBook Pro for work.  Was excited to find AltTab when I found the switching between windows on MacOS to be oddly limited.

Unfortunately, my company requires us to have SentinelOne installed for malware blocking.

SentinelOne quarantined, and therefore broke, AltTab after I gave it one of the needed permissions.  It popped up a message saying something like persistence_deception was detected.

This effectively means I can't use AltTab on my new Mac :(

Any workaround for this?

It's MacOS 13.4.1 on Apple Silicon by the way.

**Comment by lwouis** [comment]
AltTab is a legitimate app which is more cautious about user privacy and resources than most big apps.

I'm not sure why your company's anti-malware software blocked AltTab. If you provide me with more specifics, i can try and advise. As it is, i have no idea why it would think AltTab is to be blocked.

Thank you

**Comment by Davidian1024** [comment]
I'm definitely pro open source, which is why I was happy to find AltTab.  I'd like to try to troubleshoot this if possible.

I tried installing it again to see if I could get more detail about what happens.

I downloaded a copy of the latest release: AltTab-6.60.0.zip
I then extracted the contents of that zip, and ran AltTab by double clicking on it within a finder window.
I got a pop-up saying "this was downloaded from the internet.  are you sure you want to run it?"  I clicked yes, run the app.
I got another pop-up, this time from AltTab I think, saying something like: "I can move myself to the applications folder.  would you like me to do this?"  I clicked yes to this as well.
This is when SentinelOne threw it's popup window.
It said "threat detected. detected malicious running process.  process name: persistence_deception"
I can also see that it's quarantined files, though I can't tell which ones.

Any ideas?

**Comment by sapoepsilon** [comment]
Do you have sudo access to your machine?

**Comment by Davidian1024** [comment]
I do

**Comment by Davidian1024** [comment]
For what it's worth, I found this:  https://github.com/koekeishiya/yabai/issues/1233
Sounds like this other yabai app is also getting blocked by SentinelOne in a similar way.

**Comment by sapoepsilon** [comment]
> I do

What happens if you move alt tab using terminal and sudo ? 

```
sudo mv alt_tab_filename.app /Applications
```

**Comment by sapoepsilon** [comment]
> For what it's worth, I found this:  https://github.com/koekeishiya/yabai/issues/1233
> Sounds like this other yabai app is also getting blocked by SentinelOne in a similar way.

Can you give a permission in SentinelOne to alttab?

**Comment by Davidian1024** [comment]
I think I'd have to get my company's IT department to grant permissions to AltTab.  I considered pursuing that, but it will take some time.

Manually moving the extracted AltTab.app folder from Downloads to /Applications seems to work fine.  Nothing happens.
But, then when I try to start it from the Applications menu I get the same series of pop-ups.  First, do I trust this internet downloaded app? yet.  do I want AltTab moved to Applications?  I say yes here again, but this seems to fail.

**Comment by sapoepsilon** [comment]
> I think I'd have to get my company's IT department to grant permissions to AltTab. I considered pursuing that, but it will take some time.
> 
> Manually moving the extracted AltTab.app folder from Downloads to /Applications seems to work fine. Nothing happens. But, then when I try to start it from the Applications menu I get the same series of pop-ups. First, do I trust this internet downloaded app? yet. do I want AltTab moved to Applications? I say yes here again, but this seems to fail.

Another thing you could try, would be launching alttab with a _right click -> Open_ 

If that won't work, then you could try building the app yourself and signing it with your own account. That probably would bypass SentinelOne restriction for the time being.

**Comment by lwouis** [comment]
@Davidian1024 from the steps you described, I can share a few comments:

* You should run apps from the Applications folder. That's what macOS expects. There are subtle ways in which issues can arise if you run apps from another location.
* This SentinelOne app categorized AltTab with `persistence_deception` after AltTab tried to move itself to the Applications folder. I can understand the logic. They think the app is deceiving you by moving itself there, so that it can persist on your machine. They must have seen bad apps do such a pattern, and blocked it

> Manually moving the extracted AltTab.app folder from Downloads to /Applications seems to work fine. Nothing happens.
But, then when I try to start it from the Applications menu I get the same series of pop-ups. First, do I trust this internet downloaded app? yet. do I want AltTab moved to Applications? 

AltTab only asks to be moved to the Applications folder if it's not there already. You said you moved it there manually, and that it asked to move itself there. It seems to imply that you did not, in fact, move it there. Perhaps you moved it to the `~/Applications` folder, not the `/Applications` folder?

**Comment by Davidian1024** [comment]
I am fairly sure I moved it to /Applications.

I made a few tries at this, different ways.

Each time it was definitely in /Applications.  I had to first delete old copies of it before I could do that each time.  SentinelOne seemed to quarantine some of the files from within that folder, but it left most of them, leaving the app broken.

Unfortunately, it looks like I shouldn't have kept pushing it with SentinelOne.  On my most recent attempt, it decided to not only quarantine AltTab, but most of what I had been running within tmux, which is where I was doing all this.  That includes things I depend on like kubectl, alot of python stuff, and tmux itself.

I'm emailing my IT department now.  This is not the first time SentinelOne has been a thorn in my side...
